src/components/lib/kernel/cos_kernel_api.c

```diff
@@ -869,30 +985,62 @@ cos_comp_alloc_shared(struct cos_compinfo *ci, pgtblcap_t ptc, vaddr_t entry, st
 	return 0;
 }
 
+static pgtblcap_t
+__alloc_pgtbl_secondlvl(struct cos_compinfo *ci, vaddr_t heap_ptr)
+{
+	pgtblcap_t pte_cap;
+
+	pte_cap = __bump_mem_expand_intern(__compinfo_metacap(ci), ci->pgtbl_cap, heap_ptr, 0, 0);
+	assert(pte_cap);
+
+	/* 
+	 * this should only be called during component initialization
```

Comment:
I don't understand this comment. Is the argument that this line is, in fact, not needed?